SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -110,47 +110,57 @@ Interest InterestFactory::CreateServerWorkerInterest(const UAbstractLBStrategy*
 
 	Query ServerQuery{};
 
-	// Ensure server worker receives AlwaysRelevant entities.
-	ServerQuery.ResultComponentIds = ServerNonAuthInterestResultType;
-	ServerQuery.Constraint = CreateAlwaysRelevantConstraint();
-	AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, ServerQuery);
-
 	// Workers have interest in all system worker entities.
 	ServerQuery = Query();
 	ServerQuery.ResultComponentIds = SchemaResultType{ SpatialConstants::WORKER_COMPONENT_ID };
 	ServerQuery.Constraint.ComponentConstraint = SpatialConstants::WORKER_COMPONENT_ID;
 	AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, ServerQuery);
 
-	// and server worker entities
+	// And an interest in all server worker entities.
 	ServerQuery = Query();
 	ServerQuery.ResultComponentIds =
 		SchemaResultType{ SpatialConstants::SERVER_WORKER_COMPONENT_ID, SpatialConstants::GDK_KNOWN_ENTITY_TAG_COMPONENT_ID };
 	ServerQuery.Constraint.ComponentConstraint = SpatialConstants::SERVER_WORKER_COMPONENT_ID;
 	AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, ServerQuery);
 
-	// Add a self query to ensure we see the well known entity tag.
-	Query AuthoritySelfQuery = {};
-	AuthoritySelfQuery.ResultComponentIds = { SpatialConstants::GDK_KNOWN_ENTITY_TAG_COMPONENT_ID };
-	AuthoritySelfQuery.Constraint.bSelfConstraint = true;
-	AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::SERVER_WORKER_COMPONENT_ID, AuthoritySelfQuery);
-
-	// Query to know about all the actors tagged with a debug component
-	if (bDebug)
-	{
-		ServerQuery = Query();
-		ServerQuery.ResultComponentIds = SchemaResultType{ SpatialConstants::GDK_DEBUG_COMPONENT_ID };
-		ServerQuery.Constraint.ComponentConstraint = SpatialConstants::GDK_DEBUG_COMPONENT_ID;
-		AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, ServerQuery);
-	}
+	// Ensure server worker receives core GDK snapshot entities.
+	ServerQuery = Query();
+	ServerQuery.ResultComponentIds = ServerNonAuthInterestResultType;
+	ServerQuery.Constraint = CreateGDKSnapshotEntitiesConstraint();
+	AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, ServerQuery);
 
 	return ServerInterest;
 }
 
-Interest InterestFactory::CreatePartitionInterest(const UAbstractLBStrategy* LBStrategy, VirtualWorkerId VirtualWorker) const
+Interest InterestFactory::CreatePartitionInterest(const UAbstractLBStrategy* LBStrategy, VirtualWorkerId VirtualWorker, bool bDebug) const
 {
 	// Add load balancing query
 	Interest PartitionInterest{};
+	Query PartitionQuery{};
+
 	AddLoadBalancingInterestQuery(LBStrategy, VirtualWorker, PartitionInterest);
+
+	// Ensure server worker receives AlwaysRelevant entities.
```

Comment:
Should this be on the server worker entity?

Reply:
The server worker entity is used for declaring interest in non-unreal type entities, where as the partition entity gives you interest on those. That's also why the debug component is added to the partition entity interest also. I don't want to receive unreal actor-entities until the partition component interest kicks in. Happy to chat about this though,